test/functional/pdo_sqlsrv/pdo_batch_query.phpt

```diff
@@ -0,0 +1,202 @@
+    // number of rows after fetching.
+    if ($cursor == 'buffered') {
+        if ($rows != $expectedRows) {
+            fatalError("Incorrect number of columns returned with buffered cursor. Expected $expectedRows rows, got $rows rows\n");
+        }
+    } else {
+        if ($before) {
+            if ($rows !== -1) {
+                fatalError("Incorrect number of rows returned before fetching with a $cursor cursor. Expected -1 rows, got $rows rows\n");
+            }
+        } else {
+            if ($cursor == 'static' or $cursor == 'keyset') {
+                if ($rows !== -1) {
+                    fatalError("Incorrect number of rows returned before fetching with a $cursor cursor. Expected -1 rows, got $rows rows\n");
+                }
+            } else {
+                if ($rows != $expectedRows) {
+                    fatalError("Incorrect number of columns returned with buffered cursor. Expected $expectedRows rows, got $rows rows\n");
+                }
+            }
+        }
+    }
+}
+
+function printCursor($element)
+{
+    $cursor = 'forward';
+    switch($element)
+    {
+        case 0:
+        echo "Testing with forward cursor...\n";
+        break;
+        case 1:
+        echo "Testing with dynamic cursor...\n";
+        $cursor = 'dynamic';
+        break;
+        case 2:
+        echo "Testing with static cursor...\n";
+        $cursor = 'static';
+        break;
+        case 3:
+        echo "Testing with keyset cursor...\n";
+        $cursor = 'keyset';
+        break;
+        case 4:
+        echo "Testing with buffered cursor...\n";
+        $cursor = 'buffered';
+        break;
+    }
+    return $cursor;
+}
+
+$conn = new PDO("sqlsrv:server=$server;database=$databaseName;", $uid, $pwd);
+$conn->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
+
+// Create and populate a table of integer types
+$tableName = 'batch_query_test';
+$columns = array(new ColumnMeta('int', $colName[0]),
+                 new ColumnMeta('tinyint', $colName[1]),
+                 new ColumnMeta('smallint',$colName[2]),
+                 new ColumnMeta('bigint', $colName[3]),
+                 new ColumnMeta('bit', $colName[4]));
+
+createTable($conn, $tableName, $columns);
+
+$inputs = array();
+
+// Insert each row. Need an associative array to use insertRow()
+for ($i = 0; $i < $expectedRows; ++$i) {
+    $inputs = array();
+    for ($j = 0; $j < $expectedResultSets; ++$j) {
+        $inputs[$colName[$j]] = $data[$j][$i];
+    }
+
+    $stmt = insertRow($conn, $tableName, $inputs);
+    unset($stmt);
+}
+
+$query = "SELECT c1_int FROM $tableName;
+          SELECT c2_tinyint FROM $tableName;
+          SELECT c3_smallint FROM $tableName;
+          SELECT c4_bigint FROM $tableName;
+          SELECT c5_bit FROM $tableName;";
+
+// Test the batch query with different cursor types
+for ($i = 0; $i < sizeof($cursors); ++$i) {
+    try {
+        $cursorType = $cursors[$i];
+        $cursor = printCursor($i);
+
```

Comment:
Here in `printCursor` you can pass `$cursorType` directly. The same goes with `checkColumnsAndRows` such that there is no need to use string variable `$cursor`